test/integration_test/s3/getObject.spec.js

```diff
@@ -0,0 +1,209 @@
+var helpers = require('../helpers');
+var AWS = helpers.AWS;
+var bucketName = helpers.sharedBucket;
+describe('download integrity', function() {
+  var params;
+  var s3;
+  function putObject(customParams, done) {
+    params = {
+      Bucket: customParams.Bucket || params.Bucket,
+      Key: customParams.Key || params.Key,
+    };
+    s3.createBucket({Bucket: params.Bucket}).promise().then(function() {
+      return s3.waitFor('bucketExists', {Bucket: params.Bucket}).promise()
+    }).then(function() {
+      return s3.putObject({
+        Bucket: params.Bucket,
+        Key: params.Key,
+        Body: customParams.Body || params.Body
+      }).promise();
+    }).then(function() {
+      return s3.waitFor('objectExists', {
+        Bucket: params.Bucket,
+        Key: params.Key,
+      }).promise();
+    }).then(function() { 
+      if (typeof done === 'function') done();
+    }).catch(function(err) {
+      throw new Error('Cannot put object: ' + err);
+      exit(1);
+    });
+  }
+
+  function delectBucket(done) {
+    s3.listObjectVersions({
+      Bucket: params.Bucket
+    }).promise().then(function(data) {
+      var removeObjectsParams = {
+        Bucket: params.Bucket,
+        Delete: {
+          Quiet: false,
+          Objects: []
+        }
+      }
+      for (var version of data.Versions) {
+        removeObjectsParams.Delete.Objects.push({
+          Key: version.Key,
+          VersionId: version.VersionId === 'null' ? null : version.VersionId,
+        });
+      }
+      return removeObjectsParams;
+    }).then(function(removeObjectsParams) {
+      return s3.deleteObjects(removeObjectsParams).promise();
+    }).then(function() {
+      return s3.waitFor('objectNotExists', {
+        Bucket: params.Bucket,
+        Key: params.Key,
+      }).promise()
+    }).then(function() {
+      return s3.deleteBucket({Bucket: params.Bucket}).promise();
+    }).then(function(data) {
+      return s3.waitFor('bucketNotExists', {Bucket: params.Bucket}).promise();
+    }).then(function() {
+      if (typeof done === 'function') done();
+    }).catch(function(err) {
+      throw new Error('Cannot delete bucket: ' + err);
+      exit(1);
+    })
+  }
+
+  before('setup s3 client with responseChecksumAlgorithm equals \'md5\'', function() {
+    s3 = new AWS.S3({responseChecksumAlgorithm: 'md5'});
+  })
+
+  beforeEach('setup bucket and object...', function(done) {
+    params = {
+      Bucket: bucketName,
+      Key: 'key',
+    };
+    putObject({Body: 'this is a test!'}, done);
+  })
+
+  afterEach('delete bucket...', function(done) {
```

Comment:
If you `createBucket` in the `before` hook, you can `deleteBucket` in the `after` hook!